static/js/message_list_data.js

```diff
@@ -170,15 +170,25 @@ class MessageListData {
         return messages.filter((message) => predicate(message));
     }
 
+    exclude_topic_muted_messages(messages) {
+        if (!this.excludes_muted_topics) {
+            return messages;
```

Comment:
If we wanted to be extra defensive here, we could return `[...messages]`, which would mean the function consistently returns a new list.  I doubt we would ever mutate the list that gets passed in to this function in a harmful way, but it's also very cheap to copy the list here.

Reply:
(I also propose to rename this function in the general comments.)